charmcraft/reactive_plugin.py

```diff
@@ -140,6 +141,26 @@ def get_build_commands(self) -> List[str]:
         return [" ".join(shlex.quote(i) for i in command)]
 
 
+def run_charm_tool(args: List[str]):
+    """Run the charm tool, log and check exit code."""
+    result_classification = "SUCCESS"
+    exc = None
+
+    try:
+        completed_process = subprocess.run(args, check=True)
+    except subprocess.CalledProcessError as call_error:
+        exc = call_error
+        if call_error.returncode < 100 or call_error.returncode >= 200:
+            result_classification = "ERROR"
+            raise
+        else:
+            result_classification = "WARNING"
+    finally:
+        emit.debug(
```

Comment:
This would need to be converted to a `print` call. 

Also, please do this small change: print the command to be executed at the beginning of this function, which gets us two benefits: *this* message will be shorter and simpler, and we will have the command that is being executed in case that it takes too long (or "forever").

Reply:
:+1: